gtest-parallel

```diff
@@ -23,8 +23,48 @@ import threading
 import time
 import zlib
 
+# Return the width of the terminal, or None if it couldn't be
+# determined (e.g. because we're not being run interactively).
+def term_width(out):
+  if not out.isatty():
+    return None
+  try:
+    p = subprocess.Popen(["stty", "size"],
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    (out, err) = p.communicate()
+    if p.returncode != 0 or err:
+      return None
+    return int(out.split()[1])
+  except (IndexError, OSError, ValueError):
+    return None
+
+# Output transient and permanent lines of text to an output file
+# object. A transient line will be truncated to the detected line
+# width, and will overwrite an immediately preceding transient line;
+# however, if a transient line is followed by a permanent line, or a
+# permanent by a transient, nothing will be overwritten. Finally, if
+# the line width can't be determined, lines are never overwritten.
```

Comment:
Can you make this comment fit into like half of the lines? Also make sure you say why you're doing it instead of explaining the code. What does this result in?